x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/FunctionRegistryTests.java

```diff
@@ -41,12 +40,12 @@ public void testNoArgFunction() {
         assertEquals(ur.source(), ur.buildResolved(randomConfiguration(), def).source());
 
         // Distinct isn't supported
-        ParsingException e = expectThrows(ParsingException.class, () ->
+        SqlIllegalArgumentException e = expectThrows(SqlIllegalArgumentException.class, () ->
```

Comment:
Why is the `ParsingException` not thrown anymore?

Reply:
Thanks for catching that! the default `def()` method to register the functions is catching `IllegalArgumentException` and throwing `ParsingException`, so now some important info (like line number/column & function name is missing). Fixing that.